cmpserver/plugin/plugin.go

```diff
@@ -128,7 +129,26 @@ func environ(envVars []*apiclient.EnvEntry) []string {
 }
 
 // GenerateManifest runs generate command from plugin config file and returns generated manifest files
-func (s *Service) GenerateManifest(ctx context.Context, q *apiclient.ManifestRequest) (*apiclient.ManifestResponse, error) {
+func (s *Service) GenerateManifest(stream apiclient.ConfigManagementPluginService_GenerateManifestServer) error {
+	workDir, env, err := cmp.ReceiveApplicationStream(stream.Context(), stream)
+	if err != nil {
+		return fmt.Errorf("generate manifest error receiving stream: %s", err)
+	}
+	defer os.RemoveAll(workDir)
```

Comment:
If error is returned the workdir is always an empty string. However there was a bug in `cmp.ReceiveApplicationStream` that wasn't properly cleaning up the workdir in case of errors. Fixed that.